### DISCUSSION TITLE
setting input registers as floats

### DISCUSSION THREAD

**Discussion by AWilson10** [pymodbus-dev/pymodbus#1783] [Help] [answered]
Does anyone know how to configure modbus tcp server to allow floats instead of ints?
I am updating the values in a callback data block

**Comment by janiversen** [comment, marked as answer]
Best suggestion is to read the documentation. Pymodbus returns 16bit REGISTERS because that is how modbus is defined. not int32, float64 etc....you need to convert the register you receive into the datatypes you want....we actually provide several examples showing how to do that.

**Comment by truschival** [comment]
You can take a look at the [payload_server.py](https://github.com/pymodbus-dev/pymodbus/blob/dev/examples/server_payload.py) - you will need a client that can decode the message.
